src/hotspot/cpu/ppc/ppc.ad

```diff
@@ -5887,6 +5919,21 @@ instruct loadConL32_Ex(iRegLdst dst, immL32 src) %{
   %}
 %}
 
+// Load 34-bit long constant using prefixed addi. No constant pool entries required.
+instruct loadConL34(iRegLdst dst, immL34 src) %{
+  match(Set dst src);
+  ins_cost(DEFAULT_COST+1);
```

Comment:
There's no predicate for >= POWER10.  I can see how this works because of the immL34 operand having its own predicate, but in later instructs, e.g. addL_reg_imm34 even though the operand is immI32, you still add the explicit predicate.

I'd rather there be an explicit POWER10 predicate in this instruct.

Reply:
If predicate is added, adlc fails with an error message: "Syntax Error: :ADLC does not support instruction chain rules with predicates"  I think addL_reg_imm34 allows predicate because it is not called from other rules.  Is it better to leave some comments?  (BTW, immI32 is only for POWER10 or higher.  POWER9 version uses immI16 or immI16hi.)

Reply:
Hmm, I'm confused.  I don't see any other reference to loadConL34 in ppc.ad.

Reply:
I wish we knew why this predicate is causing an issue, but I guess it's not important because the operand type provides sufficient limiting of the instruct.